### PR TITLE
CLI: Add init telemetry for CLI integrations

### PR DESCRIPTION
**Pull request by shilman**
## What I did

Better understand how various CLI integrations contribute to Storybook growth

#### The changes in this PR are covered in the following automated tests:

- [ ] stories
- [x] unit tests
- [ ] integration tests
- [ ] end-to-end tests

#### Manual testing

After this has been released as stable:
- [ ] Create a new project using one of the integrations with `STORYBOOK_TELEMETRY_DEBUG`
- [ ] Verify that the correct `cliIntegration` is sent

### 🦋 Canary release

This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/publish.yml) or locally with `gh workflow run --repo storybookjs/storybook publish.yml --field pr=<PR_NUMBER>`_

## Summary by CodeRabbit

* **New Features**
  * Enhanced CLI integration detection to recognize additional frameworks and invocation patterns, including direct and nested command usages.

* **Tests**
  * Expanded test coverage for CLI integration detection to validate identification across varied command arrangements and edge cases.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: code/lib/create-storybook/src/services/VersionService.test.ts [hunk 1]
```diff
@@ -131,6 +131,49 @@ describe('VersionService', () => {
...
+      expect(integration).toBe('create-rsbuild');
+    });
+
+    it('should detect "create rsbuild" with space instead of dash', () => {
+      const ancestry = [{ command: 'npm create rsbuild -- my-app' }];
+
+      const integration = versionService.getCliIntegrationFromAncestry(ancestry as any);
+
+      expect(integration).toBe('create-rsbuild');
+    });
```
[the remark this answers is not in the document]
Comment (shilman) [reply code/lib/create-storybook/src/services/VersionService.test.ts:157]: @copilot update this pull request to apply changes based on [this feedback](https://github.com/storybookjs/storybook/pull/33603#discussion_r2711080912)

**Review by valentinpalkovic — APPROVED** [review]
